CMakeLists.txt

```diff
@@ -6,25 +6,43 @@ project (Beast)
 
 set_property (GLOBAL PROPERTY USE_FOLDERS ON)
 
+set(Boost_USE_STATIC_LIBS ON)
+set(Boost_USE_MULTITHREADED ON)
+
+
 if (MSVC)
     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP /W4 /wd4100 /bigobj /D _WIN32_WINNT=0x0601 /D_SCL_SECURE_NO_WARNINGS=1 /D_CRT_SECURE_NO_WARNINGS=1")
     set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /SAFESEH:NO")
-else()
-    set(Boost_USE_STATIC_LIBS ON)
-    set(Boost_USE_MULTITHREADED ON)
-    find_package(Boost REQUIRED COMPONENTS coroutine context thread filesystem program_options system)
-    include_directories(${Boost_INCLUDE_DIRS})
-    link_directories(${Boost_LIBRARY_DIR})
 
+    # Workaround for MSVC having two boost versions - x86 and x64 on same PC in stage folders
+    if ((NOT DEFINED BOOST_ROOT) AND (DEFINED ENV{BOOST_ROOT}))
+        if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+            set(Boost_LIBRARY_DIR $ENV{BOOST_ROOT}/stage64/lib)
+        elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
+            set(Boost_LIBRARY_DIR $ENV{BOOST_ROOT}/stage/lib)
+        endif()
+    elseif(DEFINED BOOST_ROOT)
+        if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+            set(Boost_LIBRARY_DIR ${BOOST_ROOT}/stage64/lib)
+        elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
+            set(Boost_LIBRARY_DIR ${BOOST_ROOT}/stage/lib)
+        endif()
+    endif()
+else()
     set(THREADS_PREFER_PTHREAD_FLAG ON)
     find_package(Threads)
 
-    set(CMAKE_CXX_FLAGS
-      "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wpedantic")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wpedantic")
 endif()
 
-if (MINGW)
+find_package(Boost REQUIRED COMPONENTS coroutine context thread filesystem program_options system)
+include_directories(${Boost_INCLUDE_DIRS})
+link_directories(${Boost_LIBRARY_DIRS})
```

Comment:
Sorry for the delay. This worked in every combination I was able to come up with. Additionally, @vinniefalco I was able to build boost using the rippled instructions in the same directory, and successfully build rippled in every combination I could come up with, too. It would probably be worthwhile to update the rippled builds and instructions to use this method, though, so we don't get drift.